frontend/views/containers/proposals/ProposalVoteOptions.vue

```diff
@@ -0,0 +1,204 @@
+<template lang='pug'>
+.c-ctas
+  .buttons(v-if='!hadVoted || ephemeral.changingVote')
+    i18n.button.is-outlined.is-small.is-success(
+      tag='button'
+      @click='voteFor'
+      data-test='voteFor'
+    ) Vote yes
+
+    i18n.button.is-outlined.is-small.is-danger(
+      tag='button'
+      @click='voteAgainst'
+      data-test='voteAgainst'
+    ) Vote no
+
+  .c-cancel(v-else)
+    i18n.button.is-outlined.is-small(
+      tag='button'
+      v-if='ownProposal'
+      @click='cancelProposal'
+      data-test='cancelProposal'
+    ) Cancel Proposal
+    p.has-text-1(v-else data-test='voted')
+      | {{ L('You voted {voteStatus}', { voteStatus }) }}.
+      | &nbsp;
+      i18n.link(tag='button' @click='startChangingVote') Change vote.
+  .help.has-text-danger.c-error(v-if='ephemeral.errorMsg') {{ ephemeral.errorMsg }}
+</template>
+
+<script>
+import { mapGetters, mapState } from 'vuex'
+import sbp from '~/shared/sbp.js'
+import L from '@view-utils/translations.js'
+import { VOTE_FOR, VOTE_AGAINST } from '@model/contracts/voting/rules.js'
+import { TYPE_MESSAGE } from '@model/contracts/mailbox.js'
+import { oneVoteToPass, buildInvitationUrl } from '@model/contracts/voting/proposals.js'
+import { generateInvites } from '@model/contracts/group.js'
+
+export default {
+  name: 'Vote',
+  props: {
+    proposalHash: String
+  },
+  data () {
+    return {
+      ephemeral: {
+        changingVote: false
+      },
+      errorMsg: null
+    }
+  },
+  computed: {
+    ...mapState([
+      'currentGroupId'
+    ]),
+    ...mapGetters([
+      'currentGroupState',
+      'groupSettings',
+      'currentUserIdentityContract'
+    ]),
+    proposal () {
+      return this.currentGroupState.proposals[this.proposalHash]
+    },
+    currentUsername () {
+      return this.currentUserIdentityContract.attributes.name
+    },
+    voteStatus () {
+      const humanStatus = {
+        [VOTE_FOR]: L('yes'),
+        [VOTE_AGAINST]: L('no')
+      }
+      return humanStatus[this.proposal.votes[this.currentUsername]]
+    },
+    meta () {
+      return this.proposal.meta
+    },
+    type () {
+      return this.proposal.data.proposalType
+    },
+    data () {
+      return this.proposal.data.proposalData
+    },
+    hadVoted () {
+      return this.proposal.votes[this.currentUsername]
+    },
+    ownProposal () {
+      return this.currentUsername === this.proposal.meta.username
+    }
+  },
+  methods: {
+    startChangingVote () {
+      this.ephemeral.changingVote = true
+    },
+    async voteFor () {
+      this.ephemeral.changingVote = false
+      this.ephemeral.errorMsg = null
+      const isSure = confirm(L('Are you sure you want to vote yes?'))
```

Comment:
Don't need this. Should implement a cancel link so that it doesn't matter if they accidentally click

Reply:
implement a cancel click? can that be done in another PR?